selfdrive/loggerd/v4l_encoder.cc

```diff
@@ -0,0 +1,363 @@
+#include <cassert>
+#include <sys/ioctl.h>
+#include <poll.h>
+
+#include "selfdrive/loggerd/v4l_encoder.h"
+#include "selfdrive/common/util.h"
+#include "selfdrive/common/timing.h"
+
+#include "libyuv.h"
+#include "selfdrive/loggerd/include/msm_media_info.h"
+
+// has to be in this order
+#include "selfdrive/loggerd/include/v4l2-controls.h"
```

Comment:
think this should go in third_party with the linux structure, i.e. `third_party/linux/v4l2-controls.h`. perhaps we should also fix these headers in AGNOS to actually correspond to our kernel